### PR TITLE
feat: replace CAL with Chainify

### PR DESCRIPTION
**Pull request by kraikov**

### CODE REVIEW

**Review by bradleySuira — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -127,5 +101,22 @@
     "pre-commit": "1.2.2",
     "prettier": "^2.5.1"
   },
-  "packageManager": "yarn@3.1.1"
+  "packageManager": "yarn@3.1.1",
+  "resolutions": {
+    "@chainify/bitcoin": "portal:/Users/kraykov/Documents/workspaces/liquality/chainabstractionlayer-2/packages/bitcoin",
```
Comment (harshjv) [inline package.json]: is it normal to commit local dev paths to package.json?

File: src/utils/chainify.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    return false
+  }
+
+  const allowanceCallData = `0xdd62ed3e000000000000000000000000${remove0x(user)}000000000000000000000000${HTLC_ADDRESS}`
+  const allowance = await client.chain.getProvider().call({
+    data: allowanceCallData,
+    to: toAsset.contractAddress
+  })
+
+  if (new BN(allowance.toString()).gte(amount.toString())) {
```
Comment (harshjv) [inline src/utils/chainify.js]: is `.toString()` necessary here?

File: src/utils/chainify.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+const requiresApproval = async (client, asset, user, amount) => {
+  const toAsset = assets[asset]
+
+  // approve is needed only for ERC20 tokens
+  if (!toAsset.contractAddress) {
+    return false
+  }
+
+  const allowanceCallData = `0xdd62ed3e000000000000000000000000${remove0x(user)}000000000000000000000000${HTLC_ADDRESS}`
```
Comment (harshjv) [inline src/utils/chainify.js]: can you use `array.join('')` here & below for better readability?

**Review by harshjv — CHANGES_REQUESTED** [review]
Overall, PR is looking good. Once tests are green, we should be able to merge this.

File: package.json (hunk 1 above)
Comment (kraikov) [reply package.json]: no :)

**Comment by kraikov** [comment]
> Overall, PR is looking good. Once tests are green, we should be able to merge this.

I'm not sure how to make tests green as I'm not even able to run the tests from the dev branch locally.

**Review by harshjv — APPROVED** [review]
LGTM

**Pull request merged**